src/impl/diff.js

```diff
@@ -70,7 +70,12 @@ export default function(earlier, later, units, opts) {
     }
   }
 
-  const duration = Duration.fromObject(Object.assign(results, opts));
+  function isNotEmptyObject(obj) {
+    return typeof obj === 'object' && Object.getOwnPropertyNames(obj).length > 0;
+  }
+  const duration = isNotEmptyObject(results)
```

Comment:
It looks like you're right. I have to loot at this. As I remember without this lines some tests was failed. I will comment after checking.

Reply:
This condition is required, because `results` could be just empty object, and providing any options to `diff` function will cause that `Object.assign(results, opts)` will build object with opts only what is incorrect (will cause exception)